clippy_lints/src/matches.rs

```diff
@@ -664,6 +686,25 @@ fn check_match_as_ref(cx: &LateContext<'_, '_>, ex: &Expr, arms: &[Arm], expr: &
     }
 }
 
+fn check_pats_wild_match(cx: &LateContext<'_, '_>, arms: &[Arm]) {
+    for arm in arms {
+        if let PatKind::Or(ref fields) = arm.pat.kind {
+            // look for multiple fields where one at least matches Wild pattern
+            if fields.len() > 1 && fields.into_iter().any(is_wild) {
```

Comment:
Having something like `Foo::Bar | _` is fine with `#[non_exhaustive]`  enums. Please add a check for this, like in #4934

Reply:
This doesn't make any sense. If the suggestion to go from `A|B|C|_ => {}` to `_ => {}`  is correct for a normal enum it's definitely correct for a `#[non-exhaustive]` enum.

If you're using `wildcard_enum_match_arm` you should disable this lint as they'll give conflicting suggestions anyway.

Reply:
Oh yeah, you're right. 

> they'll give conflicting suggestions

I would say that the suggestions are complementary. First `wildcard_enum_match_arm` will suggest to expand the wildcard, and after that this lint suggests, that you should handle the `_` case separately.

---
In that case, I'd suggest to add a help message "consider handling `_` separately" and making the suggestion `MaybeIncorrect`. This can be done with `span_lint_and_then`.